### PR TITLE
install-cni: Generalize cilium_ready in preparation for ASM integration

### PR DESCRIPTION
**Pull request by jingyuanliang**
Didn't test.

/assign @MrHohn 
/assign @aojea 
/hold

### CODE REVIEW

File: scripts/install-cni.sh [hunk 1]
```diff
@@ -21,15 +21,16 @@ calico_ready() {
   compgen -G "/host/etc/cni/net.d/*calico*.conflist"
 }
 
-cilium_ready() {
-  echo "Running /host/home/kubernetes/bin/cilium-cni with CNI_COMMAND=VERSION"
-  # It's necessary to try running the it instead of just checking existence because
-  # Cilium's CNI installer doesn't do atomic write (write to temporary file then move).
+cni_ready() {
+  cni="$1"
+  echo "Running '/host/home/kubernetes/bin/$cni' with CNI_COMMAND=VERSION"
```
Comment (aojea) [inline scripts/install-cni.sh:26]: You know more than me, is not recommend to use the ${cni} format?

File: scripts/install-cni.sh [hunk 2]
```diff
@@ -245,7 +246,7 @@ if [ "${ENABLE_CILIUM_PLUGIN}" == "true" ]; then
   echo "Cilium plug-in is in use. Holding CNI configurations until Cilium is ready."
 
   # inotify calls back to the beginning of this script.
-  inotify /host/home/kubernetes/bin cilium-cni "$0" cilium_ready
+  inotify /host/home/kubernetes/bin cilium-cni "$0" cni_ready cilium-cni
```
Comment (aojea) [inline scripts/install-cni.sh:249]: I got lost here :)

Comment (jingyuanliang) [reply scripts/install-cni.sh:249]: inotify uses argv+3 as callback, so the callback is `"$0" cni_ready cilium-cni`.

when calling back, `"$@"` is `cni_ready cilium-cni` so we're calling `cni_ready` with `cilium-cni` as `$1`.

Comment (aojea) [reply scripts/install-cni.sh:249]: if you want a future person with that level of bash knowledge maintain this script you should write that down in the comment XD

**Review by howardjohn — APPROVED** [review]

**Comment by jingyuanliang** [comment]
Got a chance to test it.

/hold cancel

**Pull request merged**